### PR TITLE
Move includes after #define in DEM

### PR DESCRIPTION
**Pull request by acdaigneault**
### Description
Quick refactor as done in PR #1170 that moves header includes after the #define.
Some @brief has been added here and there at the same time.

### Miscellaneous (will be removed when merged)

I tried to add @brief, but did not really worked on the content of the doc since this was not the aim of this refactoring.

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Lethe documentation is up to date
- [x] The branch is rebased onto master
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent
- [x] If parameters are modified, the tests and the documentation of examples are up to date
- [x] Changelog (CHANGELOG.md) is up to date if the refactor affects the user experience or the codebase

Pull request related list:
- [x] No other PR is open related to this refactoring
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [x] If any future works is planed, an issue is opened
- [x] The PR description is cleaned and ready for merge

### CODE REVIEW

File: include/dem/dem_solver_parameters.h [hunk 1]
```diff
@@ -12,20 +12,19 @@
  * the top level of the Lethe distribution.
  *
  * ---------------------------------------------------------------------
- *
  */
+
+#ifndef lethe_dem_solver_parmmeters_h
+#define lethe_dem_solver_parmmeters_h
```
Comment (blaisb) [inline include/dem/dem_solver_parameters.h:18]:
```suggestion
#ifndef lethe_dem_solver_parameters_h
#define lethe_dem_solver_parameters_h
```

File: source/dem/CMakeLists.txt [hunk 2]
```diff
@@ -37,7 +37,7 @@ add_library(lethe-dem
   particle_wall_linear_force.cc
   particle_wall_nonlinear_force.cc
   periodic_boundaries_manipulator.cc
-  post_processing.cc
+        dem_post_processing.cc
```
Comment (blaisb) [inline source/dem/CMakeLists.txt:40]:
```suggestion
        dem_post_processing.cc
```

Can you put in alphabetical order please

File: source/dem/CMakeLists.txt [hunk 3]
```diff
@@ -90,7 +90,7 @@ add_library(lethe-dem
   ../../include/dem/particle_wall_linear_force.h
   ../../include/dem/particle_wall_nonlinear_force.h
   ../../include/dem/periodic_boundaries_manipulator.h
-  ../../include/dem/post_processing.h
+        ../../include/dem/dem_post_processing.h
```
Comment (blaisb) [inline source/dem/CMakeLists.txt:93]: Can you put in alphabetical order please

**Review by blaisb — APPROVED** [review]
2 minor things to fix and this is ready for merge :)

File: source/dem/CMakeLists.txt (hunk 3 above)
Comment (acdaigneault) [reply source/dem/CMakeLists.txt:93]: Sorry about that, I never saw the file because I use the refactor tools in CLion and forget about that detail

**Pull request merged**